controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -333,6 +338,8 @@ func (r *StorageClassRequestReconciler) reconcilePhases() (reconcile.Result, err
 		}
 
 	} else {
+		controllerutil.RemoveFinalizer(r.StorageClassRequest, v1alpha1.StorageClassRequestFinalizer)
+		r.log.Info("finalizer removed successfully")
```

Comment:
1. I don't understand why are you removing the finalizer immediately. if there is no sync condition that we need to wait for then we should not have added a finalizer in the first place.   

2. The pattern needs to be
```
if controllerutil.RemoveFinalizer(r.StorageClassRequest, v1alpha1.StorageClassRequestFinalizer) {
       r.Client.Update(....)
}